Source/Dafny/Reporting.cs

```diff
@@ -199,7 +199,14 @@ public class ConsoleErrorReporter : BatchErrorReporter {
 
         ConsoleColor previousColor = Console.ForegroundColor;
         Console.ForegroundColor = ColorForLevel(level);
-        Console.WriteLine(ErrorToString(level, tok, msg));
+        var errorLine = ErrorToString(level, tok, msg);
+        while (tok is NestedToken nt) {
+          tok = nt.Inner;
+          msg = nt.Message ?? "[Related location]";
+          errorLine += " " + msg + " (" + (TokenToString(tok) + ").");
```

Comment:
There's an unnecessary pair of parentheses (or you may choose to use the `$" ... "` form to format the appended string).